controllers/storagecluster/external_resources.go

```diff
@@ -291,6 +302,37 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) checkSCsAndCallCreateDefaultClaim(instance *ocsv1.StorageCluster) error {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.Client.List(r.ctx, storageClassList); err != nil {
+		return fmt.Errorf("failed to list storageClasses. %v", err)
+	}
+
+	storageClass := &storagev1.StorageClass{}
+	for _, sc := range storageClassList.Items {
```

Comment:
Using this form of `for range` will create an in-memory copy of the SC. It would be preferable to just capture a point to the struct in the list.

```suggestion
	for i := range storageClassList.Items {
		sc := &storageClassList.Items[i]
```

Reply:
not required since using `Get` now